### PR TITLE
Upgrade to work with GMA SDK Android version 23.0.0 and removed AppOpen orientation

### PR DESCRIPTION
**Pull request by LTPhantom**
## Description

Updated Android dependencies and classes to be compatible with GMA SDK 23.0.0. Removed deprecated calls. VERSION BUMP AND CHANGELOG UPDATE WILL COME IN FUTURE PR.

## Related Issues

- https://github.com/googleads/googleads-mobile-flutter/issues/940

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/googleads/googleads-mobile-flutter/issues
[Contributor Guide]: https://github.com/googleads/googleads-mobile-flutter/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by luo86 — APPROVED** [review]

File: packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/FlutterAdLoader.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public void loadAdManagerAppOpen(
       @NonNull AdManagerAdRequest adRequest,
       int orientation,
       @NonNull AppOpenAdLoadCallback loadCallback) {
-    AppOpenAd.load(context, adUnitId, adRequest, orientation, loadCallback);
+    AppOpenAd.load(context, adUnitId, adRequest, loadCallback);
```
Comment (jjliu15) [inline packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/FlutterAdLoader.java:62]: Can you remove orientation from the corresponding dart API?

Comment (jjliu15) [reply packages/google_mobile_ads/android/src/main/java/io/flutter/plugins/googlemobileads/FlutterAdLoader.java:62]: Orientation was also removed from iOS 11, so similar changes should be made on the iOS side as well.
This probably requires a new major version as well

File: packages/google_mobile_ads/CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,8 @@
-## Next Version
+## 5.0.0
 * Adds `MediationExtras` class to include parameters when using mediation through the implementation of `FlutterMediationExtras` in Android and `FlutterMediationExtras` in iOS
 * Deprecates `MediationNetworkExtrasProvider` and `FLTMediationNetworkExtrasProvider`.
+* Updates GMA iOS dependency to 11.1.0
```
Comment (jjliu15) [inline packages/google_mobile_ads/CHANGELOG.md:4]: Can you add a note that this removes orientation as a parameter from app open ads?

Comment (jjliu15) [reply packages/google_mobile_ads/CHANGELOG.md:4]: Also that the minSdk version is bumped for android

File: packages/google_mobile_ads/CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
-## Next Version
+## 5.0.0
```
Comment (jjliu15) [inline packages/google_mobile_ads/CHANGELOG.md:1]: Update the constants files as well

**Review by jjliu15 — CHANGES_REQUESTED** [review]

**Review by LTPhantom — COMMENTED** [review]
Updated

**Review by jjliu15 — APPROVED** [review]

**Pull request merged**